File: SoundPayiOSSDK/SoundPay.m
//
//  SoundPay.m
//  SoundPayiOSSDK
//
//  Created by Naeem Hussain on 10/08/2022.
//

#import "SoundPay.h"

@implementation SoundPay
@synthesize delegate;

+ (id)sharedInstance {
    static SoundPay *sharedInstance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        sharedInstance = [[self alloc] init];
    });
    return sharedInstance;
}

- (id)init {
    if (self = [super init]) {
        // write your code here
        
        soundPayObj = [[SoundPaySDK alloc] init];
        [[NSNotificationCenter defaultCenter] addObserver:self
                selector:@selector(notificationReceived:)
                name:@"SoundPaySdkNotificationObserver"
                object:nil];
    }
    return self;
}

- (void)dealloc
{
    [[NSNotificationCenter defaultCenter] removeObserver:self];
}

- (void) notificationReceived:(NSNotification *) notification
{
    NSDictionary *userInfo = notification.userInfo;
    if (userInfo) {
        NSString *infoType = [userInfo valueForKey:@"sdk_title"];
        NSString *stringMessage = [userInfo valueForKey:@"sdkMessage"];
        if (stringMessage) {
            [delegate messageDidReceive:stringMessage];
        }
        else {
            [delegate messageDidSent];
        }
    }

    if ([[notification name] isEqualToString:@"TestNotification"])
        NSLog (@"Successfully received the test notification!");
}

- (void) broadcastText: (const char *) textMessage
{
    [soundPayObj startPlayback:textMessage];
}

- (void) receiveBroadcastText
{
    [soundPayObj startCapture];
}

- (void) stopBroadcast
{
    [soundPayObj stopPlayback];
}

- (void) stopReceiving
{
    [soundPayObj stopCapturing];
}

@end

File: SoundPayiOSSDK/SoundPay.swift
//
//  SoundPayiOSSDK.swift
//  SoundPayiOSSDK
//
//  Created by Naeem Hussain on 03/08/2022.
//

import Foundation
import AVFoundation
import UIKit
import AudioToolbox


protocol SoundPayiOSSDKDelegate
{
    func messageDidReceive(_ string :String)
    func messageDidSend()
}

@objc public class SoundPay: NSObject {
    
    @objc static var shared = SoundPay()
    
    private var soundPayObj : SoundPaySDK!
    var delegate : SoundPayiOSSDKDelegate?
    
    private override init() {
        super.init()
        
        print("Initializer called .... ")
        soundPayObj = SoundPaySDK()
        
        NotificationCenter.default.addObserver(self, selector: #selector(self.notificationReceived(notification:)), name: Notification.Name("SoundPaySdkNotificationObserver"), object: nil)
    }
    
    deinit
    {
        NotificationCenter.default.removeObserver(self, name: Notification.Name("SoundPaySdkNotificationObserver"), object: nil)
    }
    
    @objc func notificationReceived(notification: Notification) {
        
        print("Notification have received")
        
        if let userInfo = notification.userInfo as? [String: Any] {
            let infoType = userInfo["sdk_title"] as! String
            if let sdkMessage = userInfo["string_message"] as? String
            {
                self.delegate?.messageDidReceive(sdkMessage)
            }
            else {
                delegate?.messageDidSend()
            }
        }
    }
    
    func broadCastText(_ string: String)
    {
        if string.trimmingCharacters(in: .whitespacesAndNewlines).count > 0 {
            
            self.soundPayObj.startPlayback(string.trimmingCharacters(in: .whitespacesAndNewlines))
        }
    }
    
    func receiveBroadcastText()
    {
        self.soundPayObj.startCapture()
    }
    
    func stopBroadCast()
    {
        self.soundPayObj.stopPlayback()
    }
    
    func stopReceiving()
    {
        self.soundPayObj.stopCapturing()
    }
    
}
